src/apis/EditUserTextEndpoint.js

```diff
@@ -0,0 +1,45 @@
+import apiRequest from '../apiBase/apiRequest';
+import NoModelError from '../apiBase/errors/NoModelError';
+import ValidationError from '../apiBase/errors/ValidationError';
+
+import PostModel from '../models2/PostModel';
+import CommentModel from '../models2/CommentModel';
+
+const TYPE_MAP = {
+  t1: CommentModel,
+  t3: PostModel,
+};
+
+const ENDPOINT = '/api/editusertext';
+
+export default {
+  post(apiOptions, data={}) {
+    const { thingId, text } = data;
+    if (!thingId || !text) {
+      throw new NoModelError(ENDPOINT);
+    }
+
+    const options = {
+      type: 'form',
+      query: {
+        raw_json: 1, // make sure html back from the server is un-escaped
+      },
+      body: {
+        api_type: 'json',
+        text,
+        thing_id: thingId,
+      },
+    };
+
+    return apiRequest(apiOptions, 'POST', ENDPOINT, options)
+      .then(apiResponse => {
+        const { body: { json }} = apiResponse.response;
+        if (json.errors.length) {
+          throw json.errors[0];
```

Comment:
Hey ryan, I'm not sure what you mean by "private". I don't have any strong preferences here but I thought our general practice was to use these custom error instances so it makes type checking more straightforward/explicit for [callers](https://github.com/reddit/reddit-mobile/pull/804/files#diff-9f7e2144dca9d72611cd8d6c7fde04a1R106) of these apis. It does seems like ValidationError would be the right thing though.